apps/extension/src/routes/page/onboarding/set-grpc-endpoint.tsx

```diff
@@ -3,11 +3,26 @@ import { FadeTransition } from '@repo/ui/components/ui/fade-transition';
 import { usePageNav } from '../../../utils/navigate';
 import { PagePath } from '../paths';
 import { GrpcEndpointForm } from '../../../shared/components/grpc-endpoint-form';
+import { localExtStorage } from '../../../storage/local';
+import { AppParameters } from '@penumbra-zone/protobuf/penumbra/core/app/v1/app_pb';
 
 export const SetGrpcEndpoint = () => {
   const navigate = usePageNav();
 
-  const onSuccess = (): void => {
+  // For beta testing: set the wallet block height to zero for non-mainnet chain IDs.
+  // This logic only runs after the user selects their rpc endpoint.
+  const onSuccess = async (): Promise<void> => {
+    await new Promise(resolve => setTimeout(resolve, 1000));
+
+    const storedParams = await localExtStorage.get('params');
+    if (storedParams) {
+      const parsedParams = JSON.parse(storedParams) as AppParameters;
+      if (!parsedParams.chainId.includes('penumbra-1')) {
+        await localExtStorage.set('walletCreationBlockHeight', 0);
+      }
+    }
```

Comment:
retrieves the app parameters and checks the associated chainId. 

An additional pre-step should be added, referencing https://github.com/penumbra-zone/web/pull/1853, to check if the connected wallet is the beta wallet.

Reply:
outdated